Watchdog timer

andvib opened the pull request.

Commit:
dev apis: platform: Split nRF targets

Split nrf target into nrf9160 and nrf5340 to align structure with ff tests.
Building together with TF-M will fail without this fix.

Signed-off-by: Andreas Vibeto <andreas.vibeto@nordicsemi.no>

Commit:
ff: nordic: Add IPC calls for FF target

Add IPC calls from non-secure to secure side for Firmware Framework tests

Signed-off-by: Andreas Vibeto <andreas.vibeto@nordicsemi.no>

Commit:
nordic: add watchdog timer driver

Add driver for the hardware watchdog timer
Use the watchdog timer in FF target

Signed-off-by: Andreas Vibeto <andreas.vibeto@nordicsemi.no>

SebastianBoe commented on api-tests/platform/drivers/watchdog/nrf/nrf_wdt.c:
```diff
@@ -0,0 +1,134 @@
+	/* From nRF53 documentation, the timeout in seconds is
+	 *       timeout [s] = (CRV + 1) / 32768
+	 */
+	nrf_wdt->CRV = (time_us / 1000000 * 32768) - 1;
```
When time_us == 1 we would want CRV == 0.

But it is -1 here. The minimal CRV value is 0xF.

EDIT: Strike the above, if the precendence is fixed/correct then the above is OK.

Also, I believe * has precedence over /, so we are dividing by 32k insead of multiplying by 32k. Even if I am wrong, let us add some parenthesis to not rely on obscure operator precedence.

Commit:
ff: nordic: Move wdt to target specific folder

Move WDT driver implementation to a target specific folder to allow
for different implementations for different targets.

Signed-off-by: Andreas Vibeto <andreas.vibeto@nordicsemi.no>
